testsuites/syncgateway/upgrade/testserver/sg_replicate/test_upgrade_with_sgreplicate.py

```diff
@@ -0,0 +1,611 @@
+        primary_server.upgrade_server(cluster_config=cluster_config, server_version_build=server_upgraded_version, target=server.host, cbs_platform=cbs_platform, toy_build=toy_build)
+        log_info("Adding the node back to the cluster: {}".format(server.host))
+        primary_server.add_node(server)
+        log_info("Rebalance in server: {}".format(server.host))
+        primary_server.rebalance_in(server_urls, server)
+        log_info("Checking for the server version after the rebalance : {}".format(server_upgraded_version))
+        verify_server_version(server.host, server_upgraded_version)
+        time.sleep(10)
+
+    # Upgrade the primary server
+    primary_server_services = "kv,index,n1ql"
+    log_info("Checking for the primary server version: {}".format(server_version))
+    verify_server_version(primary_server.host, server_version)
+    log_info("Rebalance out primary server: {}".format(primary_server.host))
+    secondary_server.rebalance_out(server_urls, primary_server)
+    log_info("Upgrading the primary server: {}".format(primary_server.host))
+    secondary_server.upgrade_server(cluster_config=cluster_config, server_version_build=server_upgraded_version, target=primary_server.host, cbs_platform=cbs_platform, toy_build=toy_build)
+    log_info("Adding the node back to the cluster for primary server: {}".format(primary_server.host))
+    secondary_server.add_node(primary_server, services=primary_server_services)
+    log_info("Rebalance in primary server: {}".format(primary_server.host))
+    secondary_server.rebalance_in(server_urls, primary_server)
+    log_info("Checking for the primary server version after the rebalance: {}".format(server_upgraded_version))
+    verify_server_version(primary_server.host, server_upgraded_version)
+    log_info("Upgraded all the server nodes in the cluster")
+    log_info('------------------------------------------')
+    log_info('END server cluster upgrade')
+    log_info('------------------------------------------')
+
+
+def upgrade_sync_gateway(sync_gateways, sync_gateway_version, sync_gateway_upgraded_version, sg_conf, cluster_config):
+    log_info('------------------------------------------')
+    log_info('START Sync Gateway cluster upgrade')
+    log_info('------------------------------------------')
+
+    sg_obj = SyncGateway()
+
+    for sg in sync_gateways:
+        sg_ip = host_for_url(sg["admin"])
+        log_info("Checking for sync gateway product info before upgrade")
+        verify_sync_gateway_product_info(sg_ip)
+        log_info("Checking for sync gateway version: {}".format(sync_gateway_version))
+        verify_sync_gateway_version(sg_ip, sync_gateway_version)
+        log_info("Upgrading sync gateway: {}".format(sg_ip))
+        sg_obj.upgrade_sync_gateways(
+            cluster_config=cluster_config,
+            sg_conf=sg_conf,
+            sync_gateway_version=sync_gateway_upgraded_version,
+            url=sg_ip
+        )
+
+        time.sleep(10)
```

Comment:
Add retries for checking SG is ready and active

Reply:
I remember that we decided to have  10 seconds sleep is fine based on  last time when you updated with remove sleeps. Only if it is more than a minute need retries or comment. That is why I see that this 10 seconds sleep is not removed in old Sync gateway upgrade script too for this logic. I can still add the comment on old sync gateway upgrade test too if it was accidently missed